pkg/decode/decode_protobuf.go

```diff
@@ -139,6 +142,67 @@ func RecordToMap(fr *flow.Record) config.GenericMap {
 	return out
 }
 
+func PacketToMap(pr *flow.PacketRecord) config.GenericMap {
+	out := config.GenericMap{}
+
+	if pr == nil {
+		return out
+	}
+
+	packet := gopacket.NewPacket(pr.Stream, layers.LayerTypeEthernet, gopacket.Lazy)
+	if ethLayer := packet.Layer(layers.LayerTypeEthernet); ethLayer != nil {
+		eth, _ := ethLayer.(*layers.Ethernet)
+		out["SrcMac"] = eth.SrcMAC.String()
+		out["DstMac"] = eth.DstMAC.String()
+	}
+
+	if tcpLayer := packet.Layer(layers.LayerTypeTCP); tcpLayer != nil {
+		tcp, _ := tcpLayer.(*layers.TCP)
+		out["SrcPort"] = tcp.SrcPort.String()
+		out["DstPort"] = tcp.DstPort.String()
+	} else if udpLayer := packet.Layer(layers.LayerTypeUDP); udpLayer != nil {
+		udp, _ := udpLayer.(*layers.UDP)
+		out["SrcPort"] = udp.SrcPort.String()
+		out["DstPort"] = udp.DstPort.String()
+	}
+
+	if ipv4Layer := packet.Layer(layers.LayerTypeIPv4); ipv4Layer != nil {
+		ipv4, _ := ipv4Layer.(*layers.IPv4)
+		out["SrcAddr"] = ipv4.SrcIP.String()
+		out["DstAddr"] = ipv4.DstIP.String()
+		out["Proto"] = ipv4.Protocol
+	} else if ipv6Layer := packet.Layer(layers.LayerTypeIPv6); ipv6Layer != nil {
+		ipv6, _ := ipv6Layer.(*layers.IPv6)
+		out["SrcAddr"] = ipv6.SrcIP.String()
+		out["DstAddr"] = ipv6.DstIP.String()
+		out["Proto"] = ipv6.NextHeader
+	}
+
+	if icmpv4Layer := packet.Layer(layers.LayerTypeICMPv4); icmpv4Layer != nil {
+		icmpv4, _ := icmpv4Layer.(*layers.ICMPv4)
+		out["IcmpType"] = icmpv4.TypeCode.Type()
+		out["IcmpCode"] = icmpv4.TypeCode.Code()
+	} else if icmpv6Layer := packet.Layer(layers.LayerTypeICMPv6); icmpv6Layer != nil {
+		icmpv6, _ := icmpv6Layer.(*layers.ICMPv6)
+		out["IcmpType"] = icmpv6.TypeCode.Type()
+		out["IcmpCode"] = icmpv6.TypeCode.Code()
+	}
+
+	if dnsLayer := packet.Layer(layers.LayerTypeDNS); dnsLayer != nil {
```

Comment:
do we allow DNS in packet capture mode ? pca code doesn't enable any feature including DNS

Reply:
We capture any packets as far as I saw

![image](https://github.com/netobserv/netobserv-ebpf-agent/assets/91894519/5f51385e-e718-4f9d-a4f2-62854a155f19)